htdocs/config.php

```diff
@@ -1,12 +1,23 @@
 <?php
 @date_default_timezone_set(@date_default_timezone_get());
-require_once(dirname(__FILE__) . '/config.specific.php');
 
 if (ENABLE_DEBUG) {
 	// Warn about everything when in debug mode
 	error_reporting(E_ERROR | E_WARNING | E_PARSE | E_NOTICE);
 }
 
+// Repository paths
+define('ROOT', __DIR__ . '/../');
+define('LIB', ROOT.'lib/');
+define('ENGINE', ROOT.'engine/');
+define('WWW', ROOT.'htdocs/');
+define('UPLOAD', WWW.'upload/');
+define('ADMIN', ROOT.'admin/');
+define('TOOLS', ROOT.'tools/');
+
+// Define server-specific constants
+require_once(WWW . '/config.specific.php');
```

Comment:
It'd be worth moving the config out of the web dir now, to reduce the chance of leaking the config file contents in case some setting or other is screwed up and the file gets served as text

Reply:
Sounds good to me! Is that coupled to this PR or independent?

Reply:
It can be separate